tests/backends/test_creation.py

```
@@ -88,3 +94,74 @@ def test_sql_table_creation_suffix_with_template(self):
+@mock.patch.object(OracleDatabaseCreation, '_maindb_connection', return_value=connection)
+@mock.patch('sys.stdout', new_callable=StringIO)
+@mock.patch('sys.stderr', new_callable=StringIO)
+class OracleDatabaseCreationTests(TestCase):
+
+    def _execute_raise_user_already_exists(self, cursor, statements, parameters, verbosity, allow_quiet_fail=False):
+        # raise "user already exists" only in test user creation
+        if statements and statements[0].startswith('CREATE USER'):
+            raise DatabaseError("ORA-01920: user name 'string' conflicts with another user or role name")
+
+    def _execute_raise_tablespace_already_exists(
+        self, cursor, statements, parameters, verbosity, allow_quiet_fail=False
+    ):
+        raise DatabaseError("ORA-01543: tablespace 'string' already exists")
+
+    def _execute_raise_insufficient_privileges(
+        self, cursor, statements, parameters, verbosity, allow_quiet_fail=False
+    ):
+        raise DatabaseError("ORA-01031: insufficient privileges")
+
+    def _test_database_passwd(self):
+        # mocked to avoid test user password changed
+        return connection.settings_dict['SAVED_PASSWORD']
+
+    @mock.patch.object(OracleDatabaseCreation, '_test_user_create', return_value=False)
+    def test_create_test_db(self, *mocked_objects):
+        creation = OracleDatabaseCreation(connection)
+        # test database creation raises "tablespace already exists"
+        with mock.patch.object(
+            OracleDatabaseCreation, '_execute_statements', self._execute_raise_tablespace_already_exists
+        ):
+            with mock.patch('builtins.input', return_value='no'):
+                with self.assertRaises(SystemExit):
+                    # refusal to delete test tablespace should raise SystemExit
+                    creation._create_test_db(verbosity=0, keepdb=False)
+            # ignore "tablespace already exists" error when keepdb is on
```

... error is ignored when...